### DISCUSSION TITLE
Key binding

### DISCUSSION THREAD

**Discussion by AustinThoughtLab** [jspsych/jsPsych#826] [Q&A] [answered]
Is there a way to bind keys within an experiment? For instance, in an experiment where participants press F or J depending on stimulus length, is it possible to indicate at the beginning of the experiment that F = short and J = long or vice versa?

**Comment by jodeleeuw** [comment, marked as answer]
I'm guessing that you mean in the experiment data file so that it is clear what each key means? Is that right?

If so, you could use:

```js
jsPsych.data.addProperties({
	short_key: 'f',
	long_key: 'j'
})
```

This will add two columns to your data for every trial indicating the mapping for short and long.

**Reply by jodeleeuw** [reply]
[Documentation for jsPsych.data.addProperties() here](https://www.jspsych.org/core_library/jspsych-data/#jspsychdataaddproperties)

**Reply by AustinThoughtLab** [reply]
No, I mean that in an experiment with two counterbalancing conditions, one being F = short, J = long and the other being J = short, F = long, is there a way to assign the keys to the appropriate condition while the experiment is running? Currently we have two scripts for the two counterbalancing versions but I would like to create a single version. Ideally, there would be a prompt such as "Press the short key" and a participant could press F or J to indicate the short one, or else a forced choice or drop down menu to choose from.

**Reply by jodeleeuw** [reply]
You could combine the method I suggested above with a trial to have it happen at run time:

```js
var key_balancing_trial = {
	type: 'html-keyboard-response',
  	choices: ['f','j'],
  	stimulus: '<p>Press the short key</p>',
  	on_finish: function(data){
    		if(data.key_press == 70) {
			jsPsych.data.addProperties({short_key: 'f', long_key: 'j'});
		} else {
			jsPsych.data.addProperties({short_key: 'j', long_key: 'f'});
		}
	}
}
```
